sockeye/train.py

```diff
@@ -56,6 +58,12 @@ def _build_or_load_vocab(existing_vocab_path: Optional[str], data_path: str, num
         vocabulary = sockeye.vocab.vocab_from_json(existing_vocab_path)
     return vocabulary
 
+def _dict_difference(dict1: Dict, dict2: Dict):
```

Comment:
how about:
diff = set(dict1.keys()) - set(dict2.keys())

maybe you don't need a function then anymore